app/bundles/CampaignBundle/Model/EventModel.php

```diff
@@ -1905,16 +1908,18 @@ public function invokeEventCallback($event, $settings, $lead = null, $eventDetai
      */
     public function checkEventTiming($action, \DateTime $parentTriggeredDate = null, $allowNegative = false)
     {
-        $now = new \DateTime();
-
-        $this->logger->debug('CAMPAIGN: Check timing for '.ucfirst($action['eventType']).' ID# '.$action['id']);
-
         if ($action instanceof Event) {
             $action = $action->convertToArray();
         }
+        $campaign = (array)$action['campaign'];
+        $campaignId = $campaign['id'];
```

Comment:
According to this PHP bug: https://bugs.php.net/bug.php?id=66961 - we shouldn't rely on `convertToArray()` (which in turn relies on `get_object_vars()`), because it changes original object with all of its properties, so once converting to array we can't refer to it as an object in any other piece of code after this "harmless" `convertToArray()` call. I don't say it's already a bug - I am just warning it is a potential source of bugs.